client/src/components/Registration.js

```diff
@@ -1,32 +1,132 @@
 import React from 'react'
 
-const Schedule = () => (
-  <div className="col-6">
-    <form>
-      <div className="form-group">
-        <label for="firstName">First Name</label>
-        <input type="text" className="form-control" id="firstName" placeholder="Enter your first name" required="required"/>
-      </div>
-      <div className="form-group">
-        <label for="middleName">Middle Name</label>
-        <input type="text" className="form-control" id="middleName" placeholder="Enter your middle name"/>
-      </div>
-      <div className="form-group">
-        <label for="lastName">Last Name</label>
-        <input type="text" className="form-control" id="lastName" placeholder="Enter your last name" required="required"/>
-      </div>
-      <div className="form-group">
-        <label for="email">Email address</label>
-        <input type="email" className="form-control" id="email" aria-describedby="emailHelp" placeholder="Enter email"/>
-        <small id="emailHelp" className="form-text text-muted">We'll never share your email with anyone else.</small>
-      </div>
-      <div className="form-group">
-        <label for="exampleInputPassword1">Password</label>
-        <input type="password" className="form-control" id="exampleInputPassword1" placeholder="Password"/>
+var myHeaders = new Headers();
+myHeaders.append('Content-Type', 'application/json');
+
+const validInput = "form-control is-valid";
+const invalidInput = "form-control is-invalid";
+
+class RegForm extends React.Component {
+  
+  constructor(props) {
+    super(props);
+    this.state = {
+      user: {}, 
+      isRegSuccess: false, 
+      inputValid: {},
+      inputTouched: {
+        firstName: false,
+        middleName: false,
+        lastName: false
+      }
+    };
+    this.handleChange = this.handleChange.bind(this);
+    this.handleBlur = this.handleBlur.bind(this);
+    this.handleSubmit = this.handleSubmit.bind(this);
+  }
+  
+  handleChange(event) {
+    const regex = /^[^а-яА-Я\s\W]+$/;
+    const target = event.target;
+    const value = target.value;
+    const name = target.name;
+    regex.test(value) ? 
```

Comment:
maybe would be better to replace with this.setState({inputValid: {...this.state.inputValid, [name]: regex.test(value)}}) ?